File: C++/prime-number-of-set-bits-in-binary-representation.cpp
class Solution {
public:
    int countPrimeSetBits(int left, int right) {
        int count = 0;
        set<int> primes({2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31});
        for (int n = left; n <= right; n++) {
            int bits = countBits(n);
            if (primes.count(bits)) {
                count++;
            }
        }
        return count;
    }
    
    int countBits(int n) {
        int count = 0;
        while (n) {
            if (n & 1) {
                count++;
            }
            n >>= 1;
        }
        return count;
    }
};

File: JavaScript/prime-number-of-set-bits-in-binary-representation.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
var countPrimeSetBits = function(left, right) {
    let count = 0
    for (let i = left; i <= right; i++) {
        let bits = 0
        for (let shift = i; shift > 0; shift >>= 1) {
            bits += shift & 1
        }
        if (bits === 2 || bits === 3 || bits === 5 || bits === 7 || bits === 11 || bits === 13 || bits === 17 || bits === 19 || bits === 23 || bits === 29 || bits === 31) {
            count++
        }
    }
    return count
};